server/endpoints/AdminActions.ts

```diff
@@ -184,13 +222,58 @@ export const resetProfessors: Endpoint<AdminProfessorsRequest> = {
 };
 
 export const reportReview: Endpoint<ReviewRequest> = {
-  guard: [body("id").notEmpty().isAscii()],
+  guard: [body("id").notEmpty().isAscii(), body("netId").notEmpty().isAscii()],
   callback: async (ctx: Context, request: ReviewRequest) => {
     try {
-      await Reviews.updateOne({ _id: request.id }, { $set: { visible: 0, reported: 1 } });
-      const course = (await Reviews.findOne({ _id: request.id })).class;
-      const res = await updateCourseMetrics(course);
-      return { resCode: res.resCode };
+      const student = await Students.findOne({ netId: request.netId });
+      if (student !== null) {
+        const { lastReported, numReported } = student;
+
+        const lastReportedTime = lastReported.getTime();
+        const oneDay = new Date().getTime() + 1 * 24 * 60 * 60 * 1000;
+
+        if (numReported >= 3 && lastReportedTime < oneDay) {
+          return {
+            resCode: 0,
+            message: "Reported too many reviews. Try again in 24 hours.",
+          };
+        }
+
+        await Reviews.updateOne(
+          { _id: request.id },
+          { $set: { visible: 0, reported: 1 } },
+        );
+        const course = (await Reviews.findOne({ _id: request.id })).class;
+        const res = await updateCourseMetrics(course);
+
+        if (lastReportedTime === oneDay) {
```

Comment:
this probably will never be `true` - I think this should be a `>` or `<` comparison (>‿‿<)